src/Shared/StaticWebAssets/ManifestStaticWebAssetFileProvider.cs

```diff
@@ -367,9 +366,58 @@ private sealed class OSBasedCaseConverter : JsonConverter<Dictionary<string, Sta
         {
             public override Dictionary<string, StaticWebAssetNode> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
             {
-                return new Dictionary<string, StaticWebAssetNode>(
-                    JsonSerializer.Deserialize<IDictionary<string, StaticWebAssetNode>>(ref reader, options)!,
-                    StaticWebAssetManifest.PathComparer);
+                var parsed = JsonSerializer.Deserialize<IDictionary<string, StaticWebAssetNode>>(ref reader, options)!;
+                var result = new Dictionary<string, StaticWebAssetNode>(StaticWebAssetManifest.PathComparer);
+                MergeChildren(parsed, result);
+                return result;
+
+                static void MergeChildren(
+                    IDictionary<string, StaticWebAssetNode> newChildren,
+                    IDictionary<string, StaticWebAssetNode> existing)
+                {
+                    foreach (var (key, value) in newChildren)
+                    {
+                        if (!existing.TryGetValue(key, out var existingNode))
+                        {
+                            existing.Add(key, value);
```

Comment:
Just point out that while this fix might look a bit "involved", we'll take this branch 99.99% of the time

Reply:
The logic looks totally reasonable to me.

There might be a case for moving it into some `static` method on `StaticWebAssetNode` itself (e.g., called `CombineMatchingPaths` or `NormalizePaths` or similar. The reason is that, any time anyone adds any new properties to `StaticWebAssetNode`, they would have to update this logic to know how to merge those new properties. But how would they discover that this logic exists? Unless it's in the same place where they have to add the new properties, chances are they won't realise they have to do that.

Reply:
I see what you mean.

I'm going to leave it as is for the time being, mostly because I think it's unlikely we'll need to add more properties to it, since it would mean we are doing a big change again (and I would expect at that point to be covered by a test involving casing).